c/kastore.c

```diff
@@ -527,8 +535,13 @@ kastore_insert_all(kastore_t *self, kastore_t *other)
 
     for (j = 0; j < other->num_items; j++) {
         item = other->items[j];
-        ret = kastore_put(
-            self, item.key, item.key_len, item.array, item.array_len, item.type, 0);
+        if (other->flags && KAS_PUT_BORROWS_ARRAY) {
+            ret = kastore_put(
```

Comment:
I suspect you meant to call bput here? I don't think it's worth worrying about this here as it's a static method, and only gets called in a very specific case anyway.